### DISCUSSION TITLE
Prism regions not found using Xam community Toolkit TabView (Templated View)

### DISCUSSION THREAD

**Discussion by LeoJHarris** [PrismLibrary/Prism#2516] [Xamarin.Forms]
Hi there I am using the Xamarin community toolkit's TabView and in my TabViewItem I have a Region much like the below

```csharp
<xct:TabView
            TabStripPlacement="Top">
            <xct:TabViewItem
                BadgeTextColor="White"
                FontSize="14"
                TextColor="Black"
                TextColorSelected="Black">
                <ContentView
                        x:Name="RefillsRegion"
                        prism:RegionManager.RegionName="RefillsRegion" />
            </xct:TabViewItem>
...

</xct:TabView>
```

I am getting a Prism region not found exception, so when I take the ContentView out of the TabView it works fine. TabView being a Templated View I can see how this would be problematic but is there some work around for this? I thought this would be a good case to put regions to good use but I can understand why this might not immediately work. Any suggestions appreciated 🙏

**Comment by chowarth** [comment]
I'm seeing the same when using the XCT `StateLayout`. We have a `Loading` state and a `Success` state; the region container is a part of the `Success` `xct:StateView`. Once loading has completed we set the state to `Success` and we call `_regionManager.RequestNavigate("Container", "Region", parameters);` - this fails with the following error: **"The region manager does not contain the {0} region."**.

I've not had chance to really debug it to find out what's going on, but the error is thrown [here](https://github.com/PrismLibrary/Prism/blob/master/src/Forms/Prism.Forms.Regions/Regions/RegionCollection.cs#L43) which makes me believe that the region was never registered with the `RegionManager.Regions` collection.

**Reply by LeoJHarris** [reply]
@chowarth Ive since adopted the Xamarin community toolkit lazy view you can check this out from the discussion here to see how it was implemented https://github.com/xamarin/XamarinCommunityToolkit/discussions/1529 (it might be helpful to you) in the case of the tabviews I still separated these into their own view models per tabview and lazy load them separately which is what regions is doing..... the only issue being is the StateLayout seems to be having an issue in recognising the state change and as such the UI doesn't reflect this.... I need follow up on the thread but haven't found the time yet.

**Comment by HASSANRAGAI** [comment]
@LeoJHarris I don't know if you still want to solve it   I had the same issue and I edited the solution and you don't have to set the ViewModel manual 
see this repo and feel free to help me if there are any edits

https://github.com/ZHEAISNSAABN/ZhEaIsNsAaBn.Xamarin/tree/master/Src/ZhEaIsNsAaBn.Xamarin/LazyView
